Lib/pdb.py

```diff
@@ -128,6 +129,35 @@ def __repr__(self):
         return self
 
 
+class ScriptTarget(str):
+    def __new__(cls, val):
+        # Mutate self to be the "real path".
+        res = super().__new__(cls, os.path.realpath(val))
+
+        # Store the original path for error reporting.
+        res.orig = val
+
+        return res
+
+    def check(self):
+        if not os.path.exists(self):
```

Comment:
Previously, the existence check was done on `mainpyfile` before it was mutated to a "real path". Now the existence check is the real path. If compatibility is needed, this line could be changed to:

```suggestion
        if not os.path.exists(self.orig):
```

But in my opinion, the existence check on either form is equivalent, and the form presented is more readable and distracts less from the primary intention for `self.orig` (the error message).

Reply:
It's not equivalent for anonymous pipes created by process substitution (see GH-142315).